### ISSUE TITLE
wk06-hw - trouble hiding outliers in boxplot (plot 2)

### ISSUE THREAD

**Issue by clumb** [UM-R-for-EnvSci-Registered-Student-2021/General_Discussion#8]
Hi,

I'm trying to hide the outliers when I plot the chloride concentrations using geom_boxplot. I can't seem to get outlier.shape = NA to work (the R Documentation says to use outlier.shape = NA). Here's the code I tried:

plot_2 <- ditch_data_long %>% 

  filter(parameter == "chloride") %>% 
  
  ggplot(aes(x = site, y = measure, colour = site)) +
  ggplot2::geom_boxplot(outlier.shape = NA) +
  geom_point(aes(x = site, y = measure, colour = site)) +
  geom_jitter(aes(x = site, y = measure, colour = site), width = 0.2, height = 0, alpha = 0.5) + 
  labs(x = "", y = "Chloride concentration")
  
I also tried 

 ggplot() +
  geom_boxplot(aes(x = site, y = measure, colour = site), outlier.shape = NA) +
  geom_point(aes(x = site, y = measure, colour = site)) +
  geom_jitter(aes(x = site, y = measure, colour = site), width = 0.2, height = 0, alpha = 0.5) + 
  labs(x = "", y = "Chloride concentration")    

I tried updating my packages and re-starting RStudio (clearing the environment). I also tried a number of work-arounds I found in the following post about the same issue from 2018.
https://github.com/tidyverse/ggplot2/issues/2505

Here's what my plot looks like:

[plot_2.pdf](https://github.com/UM-R-for-EnvSci-Registered-Student-2021/General_Discussion/files/7412551/plot_2.pdf)

Thank you!
Chelsey

**Comment by clumb** [comment]
Hmm, I thought the plot would show in my post. Let me try again.

![image](https://user-images.githubusercontent.com/90855580/138746464-3d747863-8b70-4902-b36e-061a34bab02a.png)

**Comment by aura-diaz** [comment]
Hi Chelsey! I think if you just want to hide the outliers you can change the limits of the y axis using coord_cartesian to not affect the estimates

```
  ggplot() +
        geom_boxplot(.......) +
        geom_point(.....) +
        geom_jitter(.......) +
        coord_cartesian(ylim=c(0,400),expand = FALSE) +
        labs(......)
```
Cheers,
Aura

**Comment by clumb** [comment]
Hi Aura, 

Thanks so much!
I just tried the code you sent. Strangely, the coord_cartesian didn't seem to hide outliers for my plot. 
Maybe the issue is something specific to my computer. 
I was wondering how to change the scale of the axes, so now I know how to do that! 

Thanks again!
Chelsey

**Comment by peperg** [comment]
@clumb, that is a good question! @aura-diaz Thank you for assisting as well!

Yo could, of course "zoom in" as Aura is suggesting, but then you wouldnt show those points at all , and it wouldnt help with the "middle" boxes. I know what you mean, when you add the *jitter* the outliers appear both as the points from the jitter layer as well as the points from the boxplot representing the outliers of the distribution.

I wasnt expecting you all to know that trick yet, and, in fact, if you look at my figure, that doubling is there. That being said, i am very happy you noticed the issue and found a solution to it. even if the solution seems to be causing issues.

I added the `outlier.shape = NA` option to my `geom_boxplot()` call and it did the trick. What i believe is happening in yours (looking at your second version of the plot, which i like better) is that you have both the `geom_jitter()` and the `geom_point()`. what you are seing is not the outliers from the boxplot but the points from the `geom_point()`. Once you have the jitter you dont need the original points anymore!

Let me know if that makes sense.

Pepe

**Comment by clumb** [comment]
@peperg 

Hi Pepe,

Oh! That makes sense!! Thanks so much, you are exactly right! The problem was that I had `geom_point()` **and** `geom_jitter` so I plotted the raw data twice. That's why ` coord_cartesian(ylim=c(0,400),expand = FALSE)` didn't seem to work. 

Both ways work now to hide outliers: `outlier.shape = NA` and  `coord_cartesian(ylim=c(0,400),expand = FALSE)`. I had a feeling I was missing something when the solution from the R Documentation didn't seem to work.

Thanks again 😀 

Chelsey

![image](https://user-images.githubusercontent.com/90855580/138769065-469ac46f-ade0-4e59-b011-21004f08a3f0.png)
